src/Core/Utilities/CoreHelpers.cs

```diff
@@ -76,6 +76,39 @@ internal static Guid GenerateComb(Guid startingGuid, DateTime time)
         return new Guid(guidArray);
     }
 
+    internal static DateTime DateFromComb(Guid combGuid)
+    {
+        var guidArray = combGuid.ToByteArray();
+        var daysArray = new byte[4];
+        var msecsArray = new byte[4];
+
+        Array.Copy(guidArray, guidArray.Length - 6, daysArray, 2, 2);
+        Array.Copy(guidArray, guidArray.Length - 4, msecsArray, 0, 4);
+
+        Array.Reverse(daysArray);
+        Array.Reverse(msecsArray);
+
+        var days = BitConverter.ToInt32(daysArray, 0);
+        var msecs = BitConverter.ToInt32(msecsArray, 0);
+
+        var time = new TimeSpan(days, 0, 0, 0, (int)(msecs * 3.333333));
```

Comment:
I think you would get more precision with `var time = TimeSpan.FromDays(days) + TimeSpan.FromMilliseconds(msecs * 3.333333)` because the explicit cast from `double` to `int` is probably losing you some. But we also probably don't need that.

Reply:
good idea!